TaskLayer/SearchTask/SearchTask.cs

```diff
@@ -744,6 +744,170 @@ public static void WriteMzidentml(IEnumerable<PeptideSpectralMatch> items, List<
+                         search_engine_version = GlobalVariables.MetaMorpheusVersion,
+                         precursor_mass_type = pepXML.Generated.massType.monoisotopic,
+                         fragment_mass_type = pepXML.Generated.massType.monoisotopic,
+                         search_id = 1,
+                         //generate database information
+                         search_database = new pepXML.Generated.msms_pipeline_analysisMsms_run_summarySearch_summarySearch_database
+                         {
+                             local_path = database.First().FilePath,
+                             type = pepXML.Generated.msms_pipeline_analysisMsms_run_summarySearch_summarySearch_databaseType.AA,
+                         },
+                         enzymatic_search_constraint = new pepXML.Generated.msms_pipeline_analysisMsms_run_summarySearch_summaryEnzymatic_search_constraint
+                         {
+                             enzyme = CommonParameters.DigestionParams.Protease.Name,
+                             max_num_internal_cleavages = CommonParameters.DigestionParams.MaxMissedCleavages.ToString(),
+                         },
+
+                         parameter = para.ToArray()
+                     }
+                 },
+                 }
+             };
+
+            _pepxml.msms_run_summary[0].spectrum_query = new pepXML.Generated.msms_pipeline_analysisMsms_run_summarySpectrum_query[items.Count];
+
+            var searchHits = new List<pepXML.Generated.msms_pipeline_analysisMsms_run_summarySpectrum_querySearch_resultSearch_hit>();
+            for (int i = 0; i < items.Count; i++)
+            {
+                var pepWithSetMods = items[i].CompactPeptides.First().Value.Item2.First();
+                int pepXmlMods = pepWithSetMods.allModsOneIsNterminus.Count;
+                var mods = new List<pepXML.Generated.modInfoDataTypeMod_aminoacid_mass>();
+                for (int j = 0; j < pepXmlMods; j++)
+                {
+                    var mod = new pepXML.Generated.modInfoDataTypeMod_aminoacid_mass
+                    {
+                        mass = pepWithSetMods.allModsOneIsNterminus.Values.Select(p => p.monoisotopicMass).ToList()[j],
+                        position = (pepWithSetMods.allModsOneIsNterminus.Keys.ToList()[j] - 1).ToString()
+                    };
+                    mods.Add(mod);
+                }
+
+                var searchHit = new pepXML.Generated.msms_pipeline_analysisMsms_run_summarySpectrum_querySearch_resultSearch_hit
+                {
+                    hit_rank = 1,
+                    peptide = ((items[i].BaseSequence != null) ? items[i].BaseSequence : "-"),
+                    peptide_prev_aa = pepWithSetMods.PreviousAminoAcid.ToString(),
+                    peptide_next_aa = pepWithSetMods.NextAminoAcid.ToString(),
+                    protein = ((pepWithSetMods.Protein.Accession != null) ? pepWithSetMods.Protein.Accession : "-"),
```

Comment:
we still need to deal with protein and peptide ambiguity somehow, can you find out how it is recommended to deal with this problem in pepXML?